src/dsp_tools/analyse_xml_data/extract_links_from_XML.py

```diff
@@ -93,136 +103,239 @@ def _make_graph(
     """
     g: rx.PyDiGraph = rx.PyDiGraph()  # type: ignore[type-arg] # pylint: disable=no-member
     nodes = [(id_, None, None) for id_ in all_resource_ids]
-    node_ids = [x[0] for x in nodes]
-    node_inidices = g.add_nodes_from(nodes)
-    node_id_lookup = dict(zip(node_ids, node_inidices))
-    node_index_lookup = dict(zip(node_inidices, node_ids))
-    print(f"number of nodes: {len(nodes)}")
-    resptr_edges = [(node_id_lookup[x.subject_id], node_id_lookup[x.object_id], 1) for x in resptr_instances]
-    g.add_edges_from(resptr_edges)
-    print(f"number of resptr edges: {len(resptr_edges)}")
-    xml_edges = []
+    node_indices = g.add_nodes_from(nodes)
+    node_indices = list(node_indices)  # type: ignore[assignment]
+    node_id_lookup = dict(zip(all_resource_ids, node_indices))
+    node_index_lookup = dict(zip(node_indices, all_resource_ids))
+    edges: list[tuple[int, int, ResptrLink | XMLLink]] = [
+        (node_id_lookup[x.subject_id], node_id_lookup[x.object_id], x) for x in resptr_instances
+    ]
     for xml in xml_instances:
-        xml_edges.extend(
-            [(node_id_lookup[xml.subject_id], node_id_lookup[x], xml.cost_links) for x in xml.object_link_ids]
-        )
-    g.add_edges_from(xml_edges)
-    print(f"number of xml edges: {len(xml_edges)}")
-    return g, node_index_lookup
+        edges.extend([(node_id_lookup[xml.subject_id], node_id_lookup[x], xml) for x in xml.object_link_ids])
+    g.add_edges_from(edges)
+    return g, node_index_lookup, edges, set(node_indices)
```

Comment:
4 return values are quite a lot. Could you reduce the number?

If not, could you add the missing ones to the docstring?